js-api-spec/importer.test.ts

```diff
@@ -0,0 +1,633 @@
+      mock({'dir/_other.scss': 'a {from: dir}'});
+
+      const result = compileString('@import "other";', {
+        importers: [
+          {findFileUrl: () => ({url: pathToFileURL('nonexistent/other')})},
+        ],
+        loadPaths: ['dir'],
+      });
+      expect(result.css).toBe('a {\n  from: dir;\n}');
+    });
+
+    it('wraps an error', () => {
+      expect(() => {
+        compileString('@import "other";', {
+          importers: [
+            {
+              findFileUrl() {
+                throw 'this import is bad actually';
+              },
+            },
+          ],
+        });
+      }).toThrowSassException({line: 0});
+    });
+
+    it("uses an importer's source map URL", () => {
+      mock({'dir/_other.scss': 'a {b: c}'});
+      const result = compileString('@import "other";', {
+        importers: [
+          {
+            findFileUrl: () => ({
+              url: pathToFileURL('dir/other'),
+              sourceMapUrl: new URL('u:blue'),
+            }),
+          },
+        ],
+        sourceMap: true,
+      });
+
+      expect(result.sourceMap!.sources).toInclude('u:blue');
+    });
+
+    it('rejects a non-file URL', () => {
+      expect(() => {
+        compileString('@import "other";', {
+          importers: [{findFileUrl: () => ({url: new URL('u:other.scss')})}],
+        });
+      }).toThrowSassException({line: 0});
+    });
+
+    describe('when the resolved file has extension', () => {
+      it('.scss, parses it as SCSS', () => {
+        mock({'dir/_other.scss': '$a: value; b {c: $a}'});
+        const result = compileString('@import "other";', {
+          importers: [{findFileUrl: () => ({url: pathToFileURL('dir/other')})}],
+        });
+        expect(result.css).toBe('b {\n  c: value;\n}');
+      });
+
+      it('.sass, parses it as the indented syntax', () => {
+        mock({'dir/_other.sass': '$a: value\nb\n  c: $a'});
+        const result = compileString('@import "other";', {
+          importers: [{findFileUrl: () => ({url: pathToFileURL('dir/other')})}],
+        });
+        expect(result.css).toBe('b {\n  c: value;\n}');
+      });
+
+      it('.css, allows plain CSS', () => {
+        mock({'dir/_other.css': 'a {b: c}'});
+        const result = compileString('@import "other";', {
+          importers: [{findFileUrl: () => ({url: pathToFileURL('dir/other')})}],
+        });
+        expect(result.css).toBe('a {\n  b: c;\n}');
+      });
+
+      it('.css, rejects SCSS', () => {
+        mock({'dir/_other.css': '$a: value; b {c: $a}'});
+        expect(() => {
+          compileString('@import "other";', {
+            importers: [
+              {findFileUrl: () => ({url: pathToFileURL('dir/other')})},
+            ],
+          });
+        }).toThrowSassException({
+          line: 0,
+          url: pathToFileURL('dir/_other.css'),
+        });
+      });
+    });
+
+    describe('fromImport is', () => {
```

Comment:
Should this also test `@forward` and `meta.load-css`?

Reply:
Since we're already testing those for normal importers, I think it's fine to rely on this behavior being the same.